client/tinychain/ml/mlr.py

```diff
@@ -0,0 +1,95 @@
+
+
+from tinychain.collection.tensor import einsum, Dense , Tensor
+from tinychain.ml import Layer, NeuralNet
+from tinychain.ref import After
+import tinychain as tc 
+import numpy as np
+
+
+class MultipleLinearRegression :
+    
+    @classmethod
+    def load (self,learningrate): 
+        return MultipleLinearRegression([learningrate])
+    
+    @property
+    def learningrate(self):  
+        '''
+        We use property accessor so we recall self.learningrate instead of tc.Number(self[0])
+        '''
+        return tc.Number(self[0])
+    
+    def predict(self, inputs, weights, learning = True):
+        if learning==True:
+            preds = einsum("ij,ki->kj", [weights, inputs]) 
+        else:
+            '''
+            I try to add column of ones ! 
+            '''
+ 
+            inputs_=Tensor.concatenate(1,inputs,axis=1) 
+            preds = einsum("ij,ki->kj", [weights, inputs_])            
+        return preds
+    
+    def cost_func( self, labels ,preds):
+        '''
+        I try to apply Mean Squared Error
+        '''
+        cost = Tensor.sum((labels - preds)**2)/labels.len()
+        return cost
+
+    def gradient_descent(self , inputs , weights, labels):        
+        d_weights = 2/inputs.shape[0]*(einsum("ij,ki->kj", weights, einsum("ij,ki->kj", inputs, inputs.transpose()))-
+                                                      einsum("ij,ki->kj", labels, inputs.transpose()))        
+        return d_weights
+
+
+    def update_params(self ,weights, d_weights): 
+        new_weights = weights - self.learningrate * d_weights
+        return new_weights
+
+    def fit(self, X_train, y_train,iteration):
```

Comment:
Could you use the existing `GradientDescent` optimizer for this instead?

Reply:
i will search for it i guess maybe its in nn.py ?

Reply:
It's in the `optimizer` package: https://github.com/haydnv/tinychain/blob/main/client/tinychain/ml/optimizer.py